### PR TITLE
updated issue template labels, as well as added checklist to implementation ticket

### LINKED ISSUE #6647: Clean up issue templates and include new definition of "done"
After having a meeting with the engineers we decided to update the issue template to add checklist for people to use for reference when creating an issue. 

The following templates need their headings removed:
- code_cleanup.md

The new template should include the following: 

**Before creating ticket:**
- [ ] Business requirements are verified (make sure the suggested solution will meet stakeholder’s needs)

**Ticket-Creation:**
- [ ] Make sure you have acceptance criteria in the ticket (in a checklist)
- [ ] List of all tasks that need to be completed to meet the acceptance criteria
- [ ] Ticket structured well

**Pre-Merge:**
- [ ] Code follows our code style guide [documentation WIP]
- [ ] Code is documented according to our documentation standards [Need to create]
- [ ] Unit Test Coverage [need to create coverage target and tooling]
- [ ] Integration Test Coverage [need to create guide]
- [ ] Any technical debt is documented (todo/fixme => needs an issue linked in comments)
- [ ] Any follow-up tasks are created before a PR is merged
- [ ] Code peer-reviewed by at least one other developer

**If the work had design-dev handoff, also signed off by a designer**
- [ ] Code has error handling with clear error messages 
- [ ] Incomplete features are hidden behind feature flags (for features without Update or Delete DB migrations)
- [ ] Secrets (github actions + heroku) are up to date for CI, review apps, and staging

**Post-Merge, Pre-Production Deploy:**
- [ ] Code is deployed to Staging
- [ ] Code is manually verified to work on staging (rare, but important)
- [ ] Secrets (heroku) are up to date for production in preparation for prod push
- [ ] Test management commands that may need to be run
- [ ] Release scripts are up to date (rare, but important)

**Production Deploy:**

- [ ] Production deploy successful
- [ ] Notify Stakeholders

### PR DESCRIPTION
**Pull request by danielfmiranda**
Closes #6647 
Related PRs/issues #

Removed the PR checklist as this was just a minor update to some of the issue templates. 

I updated some of the issue template's labels, as well as updated the implementation_issue file to include the new checklist with the new definition of done.

### CODE REVIEW

**Review by benhohner — APPROVED** [review]
R+ Thanks Daniel!

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name: Feature request
 about: Request new or improved functionality
 title: ''
-labels: ''
+labels: engineering, Request
```
Comment (Pomax) [inline .github/ISSUE_TEMPLATE/feature_request.md:5]: let's make this label `feature request` rather than `Request`

**Review by Pomax — CHANGES_REQUESTED** [review]
looking good, just one request before we land it

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 1 above)
Comment (danielfmiranda) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Hi @Pomax, thanks for the feedback, I have made the requested changes!

**Review by Pomax — APPROVED** [review]
Nice, let's get this in the repo =)

**Pull request merged**